ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  getNonTestNetworks,
+  getSelectedAccountsForDappConnection,
+  getSelectedInternalAccount,
+  getSelectedNetworksForDappConnection,
+  getTestNetworks,
+} from '../../../selectors';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Text,
+} from '../../../components/component-library';
+import {
+  Content,
+  Footer,
+  Header,
+  Page,
+} from '../../../components/multichain/pages/page';
+import { SiteCell } from '../../../components/multichain/pages/review-permissions-page';
+import {
+  BlockSize,
+  Display,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { PermissionNames } from '../../../../app/scripts/controllers/permissions/specifications';
+
+export const ConnectPage = ({
+  request,
+  rejectPermissionsRequest,
+  approveConnection,
+  accounts,
+  selectAccounts,
+  selectedAccountAddresses,
+  activeTabOrigin,
+  permissionsRequestId,
+}: {
+  request: any;
+  permissionsRequestId: string;
+  rejectPermissionsRequest: () => void;
+  approveConnection: (request: any) => void;
+  accounts: any[];
+  selectAccounts: (addresses: any[]) => void;
+  selectedAccountAddresses: Set<string>;
+  activeTabOrigin: string;
+}) => {
+  const t = useI18nContext();
+  const networksList = useSelector(getNonTestNetworks);
+  const selectednetworksList = useSelector(
+    getSelectedNetworksForDappConnection,
+  );
+  const testNetworks = useSelector(getTestNetworks);
+  const combinedNetworks = [...networksList, ...testNetworks];
+
+  const currentAccount = useSelector(getSelectedInternalAccount);
+  const [selectedAccounts, setSelectedAccounts] = useState(
+    selectedAccountAddresses,
+  );
+
+  const selectedAccountsForDappConnection = useSelector(
+    getSelectedAccountsForDappConnection,
+  );
+
+  // Handle account selection/deselection
+  const handleAccountClick = (address: string) => {
+    const newSelectedAccounts = new Set(selectedAccounts);
+    if (newSelectedAccounts.has(address)) {
+      newSelectedAccounts.delete(address);
+    } else {
+      newSelectedAccounts.add(address);
+    }
+    setSelectedAccounts(newSelectedAccounts);
+  };
+  // Filter networks based on chainId
+  const filteredNetworks = Array.isArray(selectednetworksList)
+    ? combinedNetworks.filter((network) =>
+        selectednetworksList.includes(network.chainId),
+      )
+    : networksList;
+
+  // Select approved accounts and networks
+  const approvedAccounts =
```

Comment:
Could this be a selector?

Reply:
We can move it in a selector but since it's based on just one condition, and we are using it one file. We can keep it here